src/CMakeLists.txt

```diff
@@ -693,28 +710,27 @@ list(APPEND TINYSPLINE_CXX_SOURCE_FILES
 include(CMakePackageConfigHelpers)
 
 # C library
-add_library(tinyspline_shared SHARED ${TINYSPLINE_C_SOURCE_FILES})
-set_target_properties(tinyspline_shared PROPERTIES
+add_library(tinyspline ${TINYSPLINE_C_SOURCE_FILES})
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(tinyspline PUBLIC TINYSPLINE_STATIC)
+endif()
+set_target_properties(tinyspline PROPERTIES
 	OUTPUT_NAME "${TINYSPLINE_C_LIBRARIES_OUTPUT_NAME}"
 	DEBUG_POSTFIX "${TINYSPLINE_DEBUG_POSTFIX}"
 	COMPILE_FLAGS "${TINYSPLINE_LIBRARY_C_FLAGS}")
-install(TARGETS tinyspline_shared
+install(TARGETS tinyspline EXPORT tinyspline
 	LIBRARY DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
 	ARCHIVE DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
-	RUNTIME DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR})
-target_include_directories(tinyspline_shared
-	PUBLIC ${TINYSPLINE_C_INCLUDE_DIR})
-add_library(tinyspline_static STATIC ${TINYSPLINE_C_SOURCE_FILES})
-set_target_properties(tinyspline_static PROPERTIES
-	OUTPUT_NAME "${TINYSPLINE_C_LIBRARIES_OUTPUT_NAME}"
-	DEBUG_POSTFIX "${TINYSPLINE_DEBUG_POSTFIX}"
-	PREFIX "lib"
-	COMPILE_FLAGS "${TINYSPLINE_LIBRARY_C_FLAGS}")
-install(TARGETS tinyspline_static
-	LIBRARY DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
-	ARCHIVE DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR})
-target_include_directories(tinyspline_static
-	PUBLIC ${TINYSPLINE_C_INCLUDE_DIR})
+	RUNTIME DESTINATION ${TINYSPLINE_INSTALL_BINARY_DIR})
+install(EXPORT tinyspline
+    DESTINATION "${TINYSPLINE_INSTALL_LIBRARY_DIR}/${TINYSPLINE_INSTALL_CMAKE_CONFIG_DIR}/tinyspline"
+        NAMESPACE "tinyspline::"
+    FILE tinyspline-targets.cmake)
+target_include_directories(tinyspline
+	PUBLIC
+	    $<BUILD_INTERFACE:${TINYSPLINE_C_INCLUDE_DIR}>
+	    $<INSTALL_INTERFACE:${TINYSPLINE_INSTALL_INCLUDE_DIR}>)
+target_link_libraries(tinyspline ${TINYSPLINE_C_STATIC_LIBRARIES})
```

Comment:
Is this call necessary? The shared library already links the required libraries (e.g, https://github.com/msteinbeck/tinyspline/blob/master/src/CMakeLists.txt#L505). Or do we need to remove:

```
set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lm")
```

and the like?

Reply:
Good catch, this should have `PRIVATE` added!

This is cmake target *magic* at work.
See https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents

`PRIVATE` for a static library means:
- a static library is not linked, only archived so the linked libraries cannot be used,
- the dependencies of a static libraries (e.g. an executable) will link to the libraries added here.

`PRIVATE` for a shared library means:
- the linked libraries are used when linking the shared library (so/dll/dylib)
- when a dependency links to the shared library, no extra libraries are added. It is private.

Reply:
By using `target_link_libraries`, the targets can be safely exported.

When you execute `make install`, a `tinyspline-targets.cmake` is installed as well.